### PR TITLE
[4.0] Smart Search when plugin is disabled

### PR DESCRIPTION
**Pull request by brianteeman**
Instead of opening  the link in a new page this PR uses a modal just as with com_redirect

after

### CODE REVIEW

File: administrator/components/com_finder/src/Controller/DisplayController.php [hunk 1]
```diff
@@ -46,6 +49,22 @@ public function display($cachable = false, $urlparams = array())
...
+				$finderPluginId   = FinderHelper::getFinderPluginId();
+				$link = HTMLHelper::_(
+					'link',
+					'#plugin' . $finderPluginId . 'Modal',
+					Text::_('COM_FINDER_CONTENT_PLUGIN'),
+					'class="alert-link" data-bs-toggle="modal" id="title-' . $finderPluginId . '"'
+				);
+				$this->app->enqueueMessage(Text::sprintf('COM_FINDER_INDEX_PLUGIN_CONTENT_NOT_ENABLED_LINK', $link), 'warning');
+			}
+		}
```
Comment (richard67) [inline administrator/components/com_finder/src/Controller/DisplayController.php:67]:
```suggestion
		}

```
Fix PHP CS.

**Comment by Kostelano** [comment]
I have tested this item :white_check_mark: successfully on 4a5b20e91995dfec6a82c2d3606f6320d3ddeaae<br /><br />After calling the plugin in a modal window and enabling it, the plugin is actually enabled, however we return to the original page without reloading the page. Those. we still have before our eyes the information that the plugin is NOT included, although we just need to refresh the page.

This is not a problem with this solution (PR), as we observe identical behavior on the REDIRECT component page when the System - Redirects plugin is disabled.

So I am submitting a successful test. <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/34449">issues.joomla.org/tracker/joomla-cms/34449</a>.</sub>

**Comment by brianteeman** [comment]
i am working on the reload issue but that will not be in this pr

**Comment by ghost** [comment]
I have tested this item :white_check_mark: successfully on 4a5b20e91995dfec6a82c2d3606f6320d3ddeaae<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/34449">issues.joomla.org/tracker/joomla-cms/34449</a>.</sub>

**Comment by alikon** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/34449">issues.joomla.org/tracker/joomla-cms/34449</a>.</sub>

File: administrator/components/com_finder/src/View/Index/HtmlView.php [hunk 2]
```diff
@@ -17,6 +17,7 @@
 use Joomla\CMS\Language\Text;
 use Joomla\CMS\MVC\View\GenericDataException;
 use Joomla\CMS\MVC\View\HtmlView as BaseHtmlView;
+use Joomla\CMS\Plugin\PluginHelper;
 use Joomla\CMS\Router\Route;
```
Comment (Quy) [inline administrator/components/com_finder/src/View/Index/HtmlView.php]: Please remove 
```
use Joomla\CMS\Factory;
use Joomla\CMS\Router\Route;
```

**Comment by drmenzelit** [comment]
Thanks

**Comment by brianteeman** [comment]
thanks

**Pull request merged**